src/dimr_bmi.c

```diff
@@ -235,7 +235,9 @@ int update(double dt) {
   printf("ZSF: %s( %g ) called.\n", __func__, dt);
 #endif
   if (dt <= 0) {
-    return DIMR_BMI_FAILURE;
+    // DIMR sends dt == 0 as a first timestep, to singal 'init'.
+    // We will silently ignore this as we are already set up.
+    return DIMR_BMI_OK;
```

Comment:
typo: singal = signal. 

Wouldn't it be better to still return a failure for dt < 0?

Reply:
Fixed it and sending failure for < 0, ==0 is a bit dangerous as dt is a double, casing to (int) in that case. Hope that works reliably.